### PR TITLE
Add linting/formatting rules

### PR DESCRIPTION
**Pull request by Brainicism**

### CODE REVIEW

**Comment by grqz** [comment]
can we follow https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions?

File: plugin/yt_dlp_plugins/extractor/getpot_bgutil_script.py [hunk 1]
```diff
@@ -6,41 +6,69 @@
...
+    )
     VERSION = __version__
 
     @classproperty(cache=True)
     def _default_script_path(self):
         home = os.path.expanduser('~')
         return os.path.join(
-            home, 'bgutil-ytdlp-pot-provider', 'server', 'build', 'generate_once.js')
+            home, 'bgutil-ytdlp-pot-provider', 'server', 'build', 'generate_once.js'
+        )
```
Comment (grqz) [inline plugin/yt_dlp_plugins/extractor/getpot_bgutil_script.py:34]: this is intentional:
> Always move trailing parentheses used for grouping/functions after the last argument. On the other hand, multi-line literal list/tuple/dict/set should closed be in a new line. Generators and list/dict comprehensions may use either style

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -8,6 +8,24 @@ concurrency:
...
+    name: Lint and format check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Ruff
+        run: pip install ruff
+      - name: Python format check (Ruff)
+        run: ruff format --check -v plugin/
+      - name: Python lint check (Ruff)
+        run: ruff check --output-format=github -v plugin/
```
Comment (grqz) [inline .github/workflows/release.yml:21]: maybe also add `autopep8`?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -8,6 +8,24 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  lint-format:
+    name: Lint and format check
+    runs-on: ubuntu-latest
```
Comment (grqz) [inline .github/workflows/release.yml:13]:
```suggestion
    runs-on: ubuntu-latest
    if: github.event_name == 'pull_request'
```

File: plugin/yt_dlp_plugins/extractor/getpot_bgutil_script.py (hunk 1 above)
Comment (Brainicism) [reply plugin/yt_dlp_plugins/extractor/getpot_bgutil_script.py:34]: All this source code changes in this PR are from the respective linters. Will see if I can copy yt-dlp's ruff config to match the behavior

File: .github/workflows/release.yml (hunk 2 above)
Comment (Brainicism) [reply .github/workflows/release.yml:22]: `ruff format` seems to fulfill the same use-case as autopep8. Do you have a preference for a certain one?

Comment (grqz) [reply .github/workflows/release.yml:22]: iirc yt-dlp's hatch fmt --check does both and sometimes autopep8 has different behaviour from ruff

File: plugin/pyproject.toml [hunk 4]
```diff
@@ -30,3 +30,23 @@ allow-direct-references = true
 
 [tool.hatch.version]
 path = "yt_dlp_plugins/extractor/getpot_bgutil.py"
+
+[tool.ruff.format]
+quote-style = "single"
+
+[tool.ruff]
+line-length = 120
```
Comment (grqz) [inline plugin/pyproject.toml:38]:
```suggestion
line-length = 100
```

Comment (grqz) [reply plugin/pyproject.toml:38]: oh its copied from yt-dlp

File: plugin/pyproject.toml [hunk 5]
```diff
@@ -30,3 +30,23 @@ allow-direct-references = true
...
+    "E501",  # line too long
+    "E731",  # do not assign a lambda expression, use a def
+    "E741",  # ambiguous variable name
+]
+select = [
+    "E",  # pycodestyle errors
+    "W",  # pycodestyle warnings
+    "F",  # pyflakes
+    "I",  # import order
+]
```
Comment (grqz) [inline plugin/pyproject.toml:52]:
```suggestion
]
[tool.autopep8]
max_line_length = 120
recursive = true
exit-code = true
jobs = 0
select = [
    "E101",
    "E112",
    "E113",
    "E115",
    "E116",
    "E117",
    "E121",
    "E122",
    "E123",
    "E124",
    "E125",
    "E126",
    "E127",
    "E128",
    "E129",
    "E131",
    "E201",
    "E202",
    "E203",
    "E211",
    "E221",
    "E222",
    "E223",
    "E224",
    "E225",
    "E226",
    "E227",
    "E228",
    "E231",
    "E241",
    "E242",
    "E251",
    "E252",
    "E261",
    "E262",
    "E265",
    "E266",
    "E271",
    "E272",
    "E273",
    "E274",
    "E275",
    "E301",
    "E302",
    "E303",
    "E304",
    "E305",
    "E306",
    "E502",
    "E701",
    "E702",
    "E704",
    "W391",
    "W504",
]
```

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -8,6 +8,25 @@ concurrency:
...
+    if: github.event_name == 'pull_request'
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Ruff
+        run: pip install ruff
+      - name: Python format check (Ruff)
+        run: ruff format --check -v plugin/
```
Comment (grqz) [inline .github/workflows/release.yml:20]:
```suggestion
        run: ruff check -v plugin/
```

**Review by grqz — APPROVED** [review]
LGTM

**Pull request merged**